operator/controller/webhook/introute/sync.py

```diff
@@ -76,16 +79,64 @@ def get_mounts(self) -> List[Mapping]:
         return volume_mounts
 
 
-def _spring_cloud_k8s_config(parent) -> Optional[Mapping]:
+def _get_updated_props_srcs(related_configmaps, props_srcs) -> List:
+    """Replaces each props_src name with the latest ConfigMap based on creationTimestamp and prop_src base name"""
+    updated_props = []
+
+    if not props_srcs or not related_configmaps:
+        _LOGGER.debug(
+            f"Either props_srcs or configmaps were missing\n Props Sources: {props_srcs} \nConfigmaps: {related_configmaps}"
+        )
+        return updated_props
+
+    for prop_src in props_srcs:
+        matching_configmaps = {}
+        for cm_name, cm in related_configmaps.items():
+            if "name" in prop_src and prop_src["name"] in cm_name:
+                matching_configmaps[cm_name] = cm
+            elif "labels" in prop_src:
+                labels = prop_src["labels"]
+                for label in labels:
+                    if (
+                        label in cm["metadata"]["labels"].keys()
+                        and labels[label] == cm["metadata"]["labels"][label]
+                    ):
+                        matching_configmaps[cm_name] = cm
+        if not matching_configmaps:
+            # No updates
+            continue
+
+        # Get the most recent Configmap
+        updated_cm = max(
```

Comment:
If I'm reading the indentations correctly, looks like the code assumes that only one ConfigMap will be matched per `prop_src`. This approach might cause issues with label selectors. Say we have two ConfigMaps `db_test` and `web_test`, both with a `{"env": "test"}` label, then having a `prop_src` spec that looks for that label should match both of those CMs, however this part of the code would only select the latest of the two.

Reply:
Yeah, this looks like an oversight. I need to treat matching configmaps based on labels a little differently.